app/src/main/java/com/example/atheneum/fragments/OwnerPageFragment.java

```diff
@@ -31,10 +56,40 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
             mainActivity = (MainActivity) getActivity();
             // set action bar title
             mainActivity.setActionBarTitle(context.getResources().getString(R.string.owner_page_title));
-
         }
 
+        retrieveBooks();
+
         return this.view;
     }
 
+    /**
+     * Retrieve books from Firebase.
+     *
+     * See: https://stackoverflow.com/questions/37902635/no-setter-field-for-warning-firebase-database-retrieve-data-populate-listview
+     */
+    public void retrieveBooks() {
+        FirebaseUser firebaseUser = FirebaseAuth.getInstance().getCurrentUser();
+        final FirebaseDatabase db = FirebaseDatabase.getInstance();
+        String dbOwnerBooksPath = getString(R.string.db_ownerCollection) + "/" + firebaseUser.getUid().toString();
+        DatabaseReference ref = db.getReference(dbOwnerBooksPath);
+
+        if (firebaseUser != null) {
+            ref.addValueEventListener(new ValueEventListener() {
```

Comment:
You might need to use a ChildEventListener the values under ownerCollection aren't books but they're keys so you have to listen for changes to those keys I think?

https://stackoverflow.com/questions/37902635/no-setter-field-for-warning-firebase-database-retrieve-data-populate-listview

Reply:
Even the code snippet you were basing it off of used child event listeners:

https://firebase.google.com/docs/database/admin/retrieve-data

Reply:
This is probably the real issue